src/precice/impl/DataContext.hpp

```diff
@@ -110,6 +111,59 @@ class DataContext {
     return mesh::locateInvalidVertexID(*_mesh, c);
   }
 
+  /**
+   * @brief Attach a just-in-time mapping to this data context and setup a corresponding MappingDataCache
+   *
+   * A just-in-time mapping ( \p justInTimeMapping ) might be shared across multiple data contexts (it's a PtrMapping).
+   * The MappingDataCache ( \p mappingCache ), however, is unique for each data context.
+   *
+   * There is no need to put this function into a derived class:
+   * The Mapping class knows the direction of the mapping and the DataContext is directly called
+   * via the API function. Since all mappings within the DataContexts (stored in a std::vector
+   * as \p _mappingContexts ) only call and steer the mapData() execution (and not the
+   * computeMapping() execution), we don't register the just-in-time mapping in the \p _mappingContexts
+   * data structures. For just-in-time mappings, only the computeMapping() part is relevant, whereas
+   * the mapData() part is essentially handled through the API function calls.
+   *
+   * @param[in] mappingContext The MappingContext which holds the mapping configuration
+   * @param[in] meshContext The MeshContext holding the relevant data to map
+   */
+  void addJustInTimeMapping(MappingContext &mappingContext, MeshContext &meshContext);
+
+  /**
+   * @brief Initializes the MappingDataCache
+   *
+   * Essentially a forward to Mapping::initializeMappingDataCache,
+   * where we provide the cache from this class itself.
+   *
+   * See Mapping::initializeMappingDataCache and the MappingDataCache documentation
+   * for more information.
+   */
+  void initializeMappingDataCache()
+  {
+    if (mappingCache) {
+      justInTimeMapping->initializeMappingDataCache(*mappingCache.get());
+      mappingCache->resetData();
+    }
+  }
+
+  /**
+   * @brief Resets the time stamp of the MappingDataCache and potentially resets the data it holds
+   *
+   * See also the impl::MappingDataCache for more details.
+   *
+   * @param resetData whether to reset the data or not
+   */
+  void invalidateMappingCache(bool resetData)
+  {
+    if (mappingCache) {
+      mappingCache->setTimeStamp(-1);
```

Comment:
? The `-1` looks to me like a hack. Why don't define a new function with a better name in the `MappingDataCache`?

```
void MappingDataCache::functionWithDescriptiveName() {
    setTimeStamp(-1);  // add a comment here what -1 means
}
```